test/resources/openshift_auth.go

```diff
@@ -151,7 +153,8 @@ func openshiftClientSetup(url, username, password string, client *http.Client, i
 		return fmt.Errorf("failed to submit login form on oauth proxy screen: %w", err)
 	}
 	//sometimes we'll reach an accept permissions page for the user if they haven't accepted these scope requests before.
-	if strings.Contains(browser.Url().Host, openshiftOauthSubdomain) {
+	//refactored, this approach assumes that if the redirected page is not the console then it looks for an approve action, previous approach would cause e2e test flakes
+	if !strings.Contains(browser.Url().Host, openshiftConsoleSubdomain) {
```

Comment:
@HubertStefanski @david-martin This will break any test that is not performing an auth request that redirects to the OpenShift console or that requires a permissions check. This won't always be the case. If two tests login to the same service that performs a permissions check, the second test will fail as the redirect page:

- Won't be the OpenShift console
- Won't be the permissions request page

It'll be the product homepage, as the permissions check was performed in the first test and isn't repeated. I seem to be able to reproduce this with Grafana pretty consistently. Tests only pass when the first Grafana test (currently a flake) "fails" so the user never performs the permissions flow.

TLDR: For Grafana, I think this will only pass when the auth flow in the "skipped" Grafana test fails.